src/battle_ai_util.c

```diff
@@ -413,7 +413,7 @@ bool32 IsDamageMoveUnusable(u32 move, u32 battlerAtk, u32 battlerDef)
     // Battler doesn't see partners Ability for some reason.
     // This is a small hack to avoid the issue but should be investigated
     if (battlerDef == BATTLE_PARTNER(battlerAtk))
-        battlerDefAbility = GetBattlerAbility(battlerDef);
+        battlerDefAbility = AI_DATA->abilities[battlerDef];
```

Comment:
I've left a comment because of this. For some reason the AI doesn't see the ability.

My suspicion is that something with the Omniscient flag is wrong but I can't figure out what. I've come across the problem several times now. It works fine ingame though.